### PR TITLE
KFRD-10 EventHandler was added

### PR DESCRIPTION
**Pull request by MedvedevBW**

### CODE REVIEW

**Comment by MedvedevBW** [comment]
@placccebo

File: src/main/scala/com/bwsw/kafka/reader/EventHandler.scala [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+/**
+  * Class is responsible for processing InputEnvelope entities from MessageQueue
+  *
+  * @tparam K type of [[org.apache.kafka.clients.consumer.ConsumerRecord]] key
+  * @tparam V type of [[org.apache.kafka.clients.consumer.ConsumerRecord]] value
+  * @tparam T type of output data which is stored in [[com.bwsw.kafka.reader.entities.OutputEnvelope[T] ]]
+  * @param messageCount count of messages for retrieve from MessageQueue
+  * @param messageQueue queue which stored InputEnvelope entities
+  */
+abstract class EventHandler[K,V,T](messageCount: Int, messageQueue: MessageQueue[K,V]) {
```
Comment (placccebo) [inline src/main/scala/com/bwsw/kafka/reader/EventHandler.scala]: reverse the order of parameters

File: src/main/scala/com/bwsw/kafka/reader/EventHandler.scala [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+package com.bwsw.kafka.reader
+
+import com.bwsw.kafka.reader.entities.OutputEnvelope
+
+/**
+  * Class is responsible for processing InputEnvelope entities from MessageQueue
```
Comment (placccebo) [inline src/main/scala/com/bwsw/kafka/reader/EventHandler.scala]: Class is responsible for processing Kafka messages wrapped in InputEnvelopes using MessageQueue

File: src/main/scala/com/bwsw/kafka/reader/EventHandler.scala [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  *
+  * @tparam K type of [[org.apache.kafka.clients.consumer.ConsumerRecord]] key
+  * @tparam V type of [[org.apache.kafka.clients.consumer.ConsumerRecord]] value
+  * @tparam T type of output data which is stored in [[com.bwsw.kafka.reader.entities.OutputEnvelope[T] ]]
+  * @param messageCount count of messages for retrieve from MessageQueue
+  * @param messageQueue queue which stored InputEnvelope entities
```
Comment (placccebo) [inline src/main/scala/com/bwsw/kafka/reader/EventHandler.scala]: count of messages  to retrieve from MessageQueue
queue containing InputEnvelope entities

File: src/main/scala/com/bwsw/kafka/reader/EventHandler.scala [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  */
+abstract class EventHandler[K,V,T](messageCount: Int, messageQueue: MessageQueue[K,V]) {
+
+  /**
+    * Override it for logic realization
```
Comment (placccebo) [inline src/main/scala/com/bwsw/kafka/reader/EventHandler.scala]: Override it for implementing custom logic

File: src/main/scala/com/bwsw/kafka/reader/EventHandler.scala [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    *
+    * @param flag boolean variable for notify high level object about some problem with message handling
```
Comment (placccebo) [inline src/main/scala/com/bwsw/kafka/reader/EventHandler.scala]: boolean variable to notify a high-level object about some problems with message handling

File: src/main/scala/com/bwsw/kafka/reader/EventHandler.scala [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    */
+  def handle(flag: => Boolean): OutputEnvelope[T]
```
Comment (placccebo) [inline src/main/scala/com/bwsw/kafka/reader/EventHandler.scala]: why do you use call-by-name?

**Review by placccebo — CHANGES_REQUESTED** [review]

File: src/main/scala/com/bwsw/kafka/reader/EventHandler.scala (hunk 6 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/kafka/reader/EventHandler.scala]: its my mistake

File: src/main/scala/com/bwsw/kafka/reader/EventHandler.scala (hunk 2 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/kafka/reader/EventHandler.scala]: ok

File: src/main/scala/com/bwsw/kafka/reader/EventHandler.scala (hunk 1 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/kafka/reader/EventHandler.scala]: ok

File: src/main/scala/com/bwsw/kafka/reader/EventHandler.scala (hunk 4 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/kafka/reader/EventHandler.scala]: ok

File: src/main/scala/com/bwsw/kafka/reader/EventHandler.scala (hunk 5 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/kafka/reader/EventHandler.scala]: ok

**Comment by MedvedevBW** [comment]
@placccebo

**Review by placccebo — APPROVED** [review]

**Pull request merged**